ext/plugins/lint-nested-interactive.js

```diff
@@ -0,0 +1,68 @@
+'use strict';
+
+/*
+ Disallows nested of interactive elements
+
+ ```
+ {{! good }}
+ <button>Click here</button> <a href="/">and a link</a>
+
+ {{! bad}}
+ <button>Click here <a href="/">and a link</a></button>
+ ```
+
+ The following values are valid configuration:
+
+   * boolean -- `true` for enabled / `false` for disabled
+ */
+
+var calculateLocationDisplay = require('../helpers/calculate-location-display');
+var buildPlugin = require('./base');
+
+module.exports = function(addonContext) {
+  var LogNestedInteractive = buildPlugin(addonContext, 'nested-interactive');
+
+  function isElementNode(node) {
+    return node.type === 'ElementNode';
+  }
+
+  LogNestedInteractive.prototype.detect = function(node) {
+    return node.type === 'Program';
+  };
+
+  LogNestedInteractive.prototype.process = function(node) {
+    node.body.filter(isElementNode).forEach(function(node) {
```

Comment:
Can you make each step on a new line?

    node.body
      .filter(isElementNode)
      forEach(function() {
        // stuff
      }, this);